models/issue_label.go

```diff
@@ -387,6 +387,90 @@ func GetLabelIDsInRepoByNames(repoID int64, labelNames []string) ([]int64, error
 		Find(&labelIDs)
 }
 
+// GetLabelsInRepoByNames get labels in a repo by label names
+func GetLabelsInRepoByNames(repoID int64, labelNames []string) ([]*Label, error) {
+	labels := make([]*Label, 0, len(labelNames))
+	return labels, x.Table("label").
+		In("name", labelNames).
+		Asc("name").
+		Cols("id", "repo_id", "org_id").
+		Find(&labels)
+}
+
+// GetLabelByIDOrName get label by id or name and repo id
+func GetLabelByIDOrName(queryID string, repoID int64) (label *Label, exist bool, err error) {
```

Comment:
I don't think this is a good idea. How to handle when a label string with a `1`?

Reply:
has added a new way in 8b8f13e0f9dd1bbae1b9c8fdef5d5522df4ab162, which allow add a query option to confirm the input value is label name not id.

example  ``/api/v1/repos/%s/%s/issues/%d/labels/12?mode=name&token=%s``